core/hash_map.h

```diff
@@ -74,13 +78,14 @@ class HashMapHahserDefault {
  * @param TKey  Key, search is based on it, needs to be hasheable. It is unique in this container.
  * @param TData Data, data associated with the key
  * @param Hasher Hasher object, needs to provide a valid static hash function for TKey
+ * @param Comparator comparator object, needs to be able to safely compare two TKey values
```

Comment:
Go doesn't cleverly work around it, it doesn't IMPLEMENT the feature we're using. In go the variant equivalent could just not be used as a key in a hash at all.

You ignored my actual point though. The comment equivalent of the go comment was already there.